### PR TITLE
implement csv-export writing maps of structs

### PR DESCRIPTION
**Pull request by mikegray**
fixing issue #2293

### CODE REVIEW

File: samples/go/csv/csv-export/exporter.go [hunk 1]
```diff
@@ -46,8 +47,16 @@ func main() {
...
+		if l, ok := hv.(types.List); ok {
+			structDesc := csv.GetListElemDesc(l, ds.Database())
+			csv.WriteList(l, structDesc, comma, os.Stdout)
+		} else if m, ok := hv.(types.Map); ok {
+			structDesc := csv.GetMapElemDesc(m, ds.Database())
+			csv.WriteMap(m, structDesc, comma, os.Stdout)
+		} else {
+			panic(fmt.Sprintf("Expected ListKind or MapKind, found %s", types.KindToString[hv.Type().Kind()]))
+		}
 	})
```
Comment (willhite) [inline samples/go/csv/csv-export/exporter.go]: What happens here if the user is exporting to a brand new dataset? Seems like that's not possible.

Comment (mikegray) [inline samples/go/csv/csv-export/exporter.go]: csv-export emits output from a dataset to stdout, not to a dataset.

Comment (willhite) [inline samples/go/csv/csv-export/exporter.go]: Of course.

**Comment by willhite** [comment]
LGTM

**Pull request merged**